shutdown.go

```diff
@@ -20,6 +20,13 @@
 
 package fx
 
+import (
+	"context"
+	"time"
+)
+
+const _receiverShutdownTimeout = 10 * time.Millisecond
```

Comment:
While I think this is a _reasonable_ timeout value, I don't think it's safe to assume that it will be for all applications.

How about just sending `context.Background()` to `receivers.Stop`, and in the future if it's necessary, we can expose a ShutdownOption to set the receivers timeout context directly from the user.

Reply:
I'll add the option, I don't think it introduces much complexity